snapcraft/plugins/rust.py

```diff
@@ -223,13 +235,27 @@ def _build_env(self):
 
         return env
 
-    def _get_toolchain(self) -> str:
+    def _get_toolchain(self) -> Optional[str]:
+        """Return the toolchain to use.
+
+        If a rust-toolchain file is present, None will be returned.
+        If a rust-toolchain file is not present and neither rust-version
+        nor rust-channel are set, then stable will be returned.
+        """
+        if self.options.source_subdir:
```

Comment:
NIT - could be simplified:

```
sourcedir = self.sourcedir
if self.options.source_subdir:
    sourcedir = os.path.join(sourcedir, self.options.source_subdir)
```

Reply:
this is a a problem with all plugins, we should fix it across instead of just here